### DISCUSSION TITLE
Any appetite here for an editor banner API proposal?

### DISCUSSION THREAD

**Discussion by gjsjohnmurray** [microsoft/vscode-discussions#659] [Extension Development QnA]
See https://github.com/microsoft/vscode/issues/151403#issuecomment-1214168194 and subsequent responses.

Are there any other extension developers here with use cases that the editor banner might be a good fit for, without it feeling spammy?

**Comment by isidorn** [comment]
I fear it might be abused if we make it into an API

**Reply by gjsjohnmurray** [reply]
Doesn't my suggestion of a built-in "Never Show Again" option mitigate this?

**Comment by matthewjamesadam** [comment]
Hello!  We definitely have a use case for this -- in some contexts our extension has to show a read-only version of a file in the user's repository, but it's important that we provide some context as to why the file is read-only.  Without showing this context the UX is very confusing.  It would be a great addition to the extension API.

**Reply by matthewjamesadam** [reply]
To provide a little more detail, it is pretty important to us that this banner be a full webview, as we provide some detailed diagnostic information, plus some actions that the user can execute to try to resolve the situation.  So a simple text label would not be sufficient for our scenario.

**Comment by burkeholland** [comment]
I don't like notifications, so I'd think banners could be just as annoying. I know that most developers would use them responsibly, but it only takes a few to make this a miserable experience.

I wonder if we should have a model where we allow the developer to accept notifications from an extension - kind of like iOS. But then again, notifications on that platform are not a great example of a good experience.

**Comment by theReynald** [comment]
Closing this due to weeks of inactivity. If you have more to add onto this discussion, feel free to reopen.
